pkg/splunk/enterprise/configuration.go

```diff
@@ -570,6 +590,20 @@ func getSplunkStatefulSet(cr enterprisev1.MetaObject, spec *enterprisev1.CommonS
 // updateSplunkPodTemplateWithConfig modifies the podTemplateSpec object based on configuration of the Splunk Enterprise resource.
 func updateSplunkPodTemplateWithConfig(podTemplateSpec *corev1.PodTemplateSpec, cr enterprisev1.MetaObject, spec *enterprisev1.CommonSplunkSpec, instanceType InstanceType, extraEnv []corev1.EnvVar) {
 
+	// Add custom ports to splunk containers
+	if spec.ServiceTemplate.Spec.Ports != nil {
```

Comment:
check for ServiceTemplate != nil and ServiceTemplate.Spec != nil ?

Reply:
These are inner structs.